test/fail_compilation/fail19095.d

```diff
@@ -0,0 +1,16 @@
+/* REQUIRED_ARGS: -de
+ * PERMUTE_ARGS:
+ * TEST_OUTPUT:
+---
+fail_compilation/fail19095.d(13): Deprecation: mismatched array lengths, 2 and 1
+fail_compilation/fail19095.d(14): Deprecation: mismatched array lengths, 2 and 1
+fail_compilation/fail19095.d(15): Deprecation: mismatched array lengths, 2 and 1
+---
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=1XXXX
+
+int[2] ARR = [1];
+static int[2] static_arr = [1];
+__gshared int[2] gshared_arr = [1];
+int[3] asoc_arr = [1:2];
```

Comment:
Wow, I didn't know that this was legal, but it really is: https://run.dlang.io/is/3ytqf6
I think this should be deprecated too!

Reply:
On the contrary, I think this is a handy way to initialize an array with mostly zeroes:
https://dlang.org/spec/arrays#static-init-static